pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -123,15 +120,15 @@ def restore_model_state(self, model: LightningModule, checkpoint) -> None:
         # restore model state_dict
         model.load_state_dict(checkpoint['state_dict'])
 
-    def restore_training_state(self, checkpoint):
+    def restore_training_state(self, checkpoint, load_optimizer_states: bool = True):
```

Comment:
Code conv is complaining that this `load_optimizer_states` function hasn't been tested. We probably want to add a test to ensure when this function is called and this variable is turned off via the training type plugin, we do not load the optimizer state.

Reply:
Yes.